pkg/dns/server.go

```diff
@@ -20,27 +22,47 @@ type DNSServer interface {
 type SimpleDNSServer struct {
 	address  string
 	resolver DNSResolver
+
+	latencyMetric    prometheus.Summary
+	resolutionMetric *prometheus.CounterVec
 }
 
-func NewDNSServer(port uint32, resolver DNSResolver) DNSServer {
+func NewDNSServer(port uint32, resolver DNSResolver, metrics core_metrics.Metrics) (DNSServer, error) {
 	handler := &SimpleDNSServer{
 		address:  fmt.Sprintf("0.0.0.0:%d", port),
 		resolver: resolver,
+		latencyMetric: prometheus.NewSummary(prometheus.SummaryOpts{
+			Name:       "dns_server",
+			Help:       "Summary of DNS Server responses",
+			Objectives: core_metrics.DefaultObjectives,
+		}),
+		resolutionMetric: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "dns_server_resolution",
+			Help: "Counter for DNS Server resolutions",
+		}, []string{"result"}),
+	}
+	if err := metrics.Register(handler.latencyMetric); err != nil {
+		return nil, err
+	}
+	if err := metrics.Register(handler.resolutionMetric); err != nil {
+		return nil, err
 	}
 	handler.registerDNSHandler()
-	return handler
+	return handler, nil
 }
 
 func (h *SimpleDNSServer) parseQuery(m *dns.Msg) {
 	for _, q := range m.Question {
 		switch q.Qtype {
 		case dns.TypeA:
-			serverLog.Info("Query for " + q.Name)
+			serverLog.V(1).Info("query for " + q.Name)
```

Comment:
V(1)? we need it?

Reply:
I think it's overkill to log every DNS request on the info level. It's like with API Server, we don't request every single request. With many services that uses DNS, logs will be spammed with "Query for..."